src/components/Api.js

```diff
@@ -0,0 +1,145 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseURL = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkTheApiResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserData() {
+    return fetch(`${this._baseURL}/users/me`, {
+      headers: this._headers,
+    })
+      .then((res) => this._checkTheApiResponse(res))
+      .catch((err) => console.error(err));
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js` after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.

Reply:
Done.